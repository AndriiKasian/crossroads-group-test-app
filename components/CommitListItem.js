import React from 'react';

import Button from '@material-ui/core/Button';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';

const CommitListItem = ({ committer, date, message, url }) => {
    const handleView = () => {
        // TODO redirect to the selected github commit page
    };

    const getFormattedDate = date => new Date(date).toLocaleString();

    return (
        <Card>
            <CardContent>
                <Typography color="textSecondary" gutterBottom>
                    {message}
                </Typography>
                <Typography variant="body2" component="p">
                    {committer} committed at {getFormattedDate(date)}
                </Typography>
            </CardContent>
            {url && <CardActions>
                <Button size="small" onClick={handleView}>View</Button>
            </CardActions>}
        </Card>
    )
};

export default CommitListItem;
