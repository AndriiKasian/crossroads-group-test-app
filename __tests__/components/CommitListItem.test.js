import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import CommitListItem from '../../components/CommitListItem'

describe('CommitListItem', () => {
    it('Should render with empty props', () => {
        const wrapper = shallow(<CommitListItem/>);

        expect(wrapper).toBeDefined();
    });

    it('Should render correctly', () => {
        const component = renderer.create(<CommitListItem/>);
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
});
