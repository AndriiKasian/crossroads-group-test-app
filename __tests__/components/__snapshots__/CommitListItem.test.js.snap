// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommitListItem Should render correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
    />
    <p
      className="MuiTypography-root MuiTypography-body2"
    >
       committed at 
      Invalid Date
    </p>
  </div>
</div>
`;
